### ISSUE TITLE
RejectedExecutionException in ReplicationInternal

### ISSUE THREAD

**Issue by Martin-Hogge** [couchbase/couchbase-lite-android#1612]
Hello !

I catch an Exception through Fabric :
```
Fatal Exception: java.util.concurrent.RejectedExecutionException: Task java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@4a0a0f0 rejected from java.util.concurrent.ScheduledThreadPoolExecutor@30dd669[Shutting down, pool size = 1, active threads = 1, queued tasks = 0, completed tasks = 17]
       at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2049)
       at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:814)
       at java.util.concurrent.ScheduledThreadPoolExecutor.delayedExecute(ScheduledThreadPoolExecutor.java:305)
       at java.util.concurrent.ScheduledThreadPoolExecutor.schedule(ScheduledThreadPoolExecutor.java:533)
       at java.util.concurrent.ScheduledThreadPoolExecutor.submit(ScheduledThreadPoolExecutor.java:635)
       at java.util.concurrent.Executors$DelegatedExecutorService.submit(Executors.java:601)
       at com.couchbase.lite.replicator.ReplicationInternal.fireTrigger(ReplicationInternal.java:235)
       at com.couchbase.lite.replicator.ReplicationInternal.triggerStopGraceful(ReplicationInternal.java:210)
       at com.couchbase.lite.replicator.PullerInternal.triggerStopGraceful(PullerInternal.java:60)
       at com.couchbase.lite.replicator.ReplicationInternal.waitForPendingFutures(ReplicationInternal.java:1930)
       at com.couchbase.lite.replicator.PullerInternal$14.run(PullerInternal.java:969)
       at java.lang.Thread.run(Thread.java:762)
```

So, if I'm right this means that their is too much threads running at the same time.
I don't really know how the `ReplicationInternal` works and handle its `threads` and I didn't manage to reproduce this issue but I prefer to report it here and see what is it about.

**Comment by hideki** [comment]
Hi @Martin-Hogge,
What version of CBL are you using?

**Comment by Martin-Hogge** [comment]
Hi @hideki I'm using the 1.4.0 version.

**Comment by hideki** [comment]
Hi @Martin-Hogge,
I believe this issue is fixed with 1.4.1.
https://github.com/couchbase/couchbase-lite-java-core/issues/1637
Could you please try `1.4.1`?
Thanks a lot!

**Comment by djpongh** [comment]
assuming 1.4.1 fixes. If you still the issue, please reopen.
